consensus/validation.go

```diff
@@ -780,12 +780,24 @@ func validateV2FileContracts(ms *MidState, txn types.V2Transaction) error {
 		case *types.V2FileContractRenewal:
 			renewal := *r
 
+			if fc.RenterPublicKey != renewal.NewContract.RenterPublicKey {
+				return fmt.Errorf("file contract renewal %v changes renter public key", i)
+			} else if fc.HostPublicKey != renewal.NewContract.HostPublicKey {
+				return fmt.Errorf("file contract renewal %v changes host public key", i)
+			}
+
+			// validate that the renewal value is equal to existing contract's value.
+			// This must be done as a sum of the outputs, since the individual payouts may have
+			// changed in an unbroadcast revision.
+			totalPayout := renewal.FinalRenterOutput.Value.Add(renewal.RenterRollover).
+				Add(renewal.FinalHostOutput.Value).Add(renewal.HostRollover)
+			existingPayout := fc.RenterOutput.Value.Add(fc.HostOutput.Value)
+			if totalPayout != existingPayout {
+				return fmt.Errorf("file contract renewal %d renewal payout (%s) does not match existing contract payout %s", i, totalPayout, existingPayout)
```

Comment:
error is a bit inconsistent with the others, which use `(%d H)`. That might predate https://github.com/SiaFoundation/core/pull/146, so perhaps we should switch them all to `%s` now?